kde/qtcurve/Portfile

```diff
@@ -1,86 +1,241 @@
+        }
+    }
+    depends_lib-append      port:gtk2
+    depends_run-append      port:perl5
+    configure.args-delete   -DQTC_ENABLE_X11:BOOL=OFF
+    configure.args-append   -DENABLE_GTK2:BOOL=ON -DQTC_ENABLE_X11:BOOL=ON \
+                            -DENABLE_QT4:BOOL=OFF -DENABLE_QT5:BOOL=OFF \
+                            -DQTC_QT4_ENABLE_KDE:BOOL=OFF -DQTC_QT5_ENABLE_KDE:BOOL=OFF \
+                            -DQTC_UTILSLIB_INFIX:STRING="-gtk2"
     post-destroot {
+        # impose Mac-like defaults
+        xinstall -m 644 ${filespath}/kdeglobals ${destroot}${prefix}/share/themes/QtCurve/gtk-2.0/kdeglobals
+    }
+}
+
+subport ${name}-extra {
+    long_description        Installs the Mac-specific default configuration plus additional \
+                            configuration presets and colour palettes for QtCurve
+    version                 ${qtc_version}
+    revision                2
+    epoch                   2
+    use_configure           no
+    installs_libs           no
+    supported_archs         noarch
+    depends_run-append      port:ciment-icons
+    fetch {}
+    checksum {}
+    extract {}
+    build {}
+    destroot {
+        xinstall -m 755 -d ${destroot}/${prefix}/share/apps/kstyle/themes
+        xinstall -m 755 -d ${destroot}/${prefix}/share/apps/color-schemes
+        xinstall -m 755 -d ${destroot}/${prefix}/etc/xdg/qtcurve
+        xinstall -m 644 ${filespath}/stylerc ${destroot}/${prefix}/etc/xdg/qtcurve/
         xinstall -m 644 ${filespath}/qtc_qtcurve-rjvb.themerc ${destroot}/${prefix}/share/apps/kstyle/themes/
         xinstall -m 644 ${filespath}/qtc_qtcurve-osx.themerc ${destroot}/${prefix}/share/apps/kstyle/themes/
+        xinstall -m 644 ${filespath}/qtc_qtcurve-osxgraphite.themerc ${destroot}/${prefix}/share/apps/kstyle/themes/
         xinstall -m 644 ${filespath}/QtCurveOSX.colors ${destroot}/${prefix}/share/apps/color-schemes/
-        xinstall -d -m 755 ${destroot}${prefix}/share/qt4/plugins
-        system "ln -s ${prefix}/lib/kde4/plugins/styles ${destroot}${prefix}/share/qt4/plugins"
+        xinstall -m 644 ${filespath}/OxygenOSXGraphite.colors ${destroot}/${prefix}/share/apps/color-schemes/
     }
 }
 
-notes-append "To access QtCurve and other KDE styles from pure Qt4 applications, execute
- ln -s ${prefix}/lib/kde4/plugins/styles ${prefix}/share/qt4/plugins
- with the appropriate privileges"
+if {${subport} ne "${name}-extra"} {
+    depends_run-append      port:${name}-extra
+    patchfiles-append       patch-systemconfig-support.diff
+    # deactivate config page(s) that don't make sense on OS X and/or don't work properly anywhere
+    patchfiles-append       patch-deactivate-config-page.diff
+    post-patch {
+        # WIP fixes for (mostly) qt5
+        # TODO: fix QPAINTER_RENDERHIT_AA_MAYBE_OFF typo upstream
+        # https://git.reviewboard.kde.org/r/129363/
+        system -W ${worksrcpath} "patch -Np1 -i ${filespath}/patch-qt5-fixes.diff"
```

Comment:
This is a patch that's under upstream review and hopefully will be accepted in the (very) near future. It's been very much in flux and git generates -P1 patches. I don't want to forget a p1->p0 conversion when I push to my repo.